bin/lint

```diff
@@ -17,42 +17,23 @@ cd "$(dirname "$0")/.."
 
 . misc/shlib/shlib.bash
 
-failed=false
-
-try() {
-    last_failed=false
-    if ! "$@"; then
-        failed=true
-        last_failed=true
-    fi
-}
-
 if [[ ! "${MZDEV_NO_SHELLCHECK:-}" ]]; then
     if ! command_exists shellcheck; then
-        printf "lint: fatal: unable to find \`shellcheck\` command on your system\n" >&2
-        printf "hint: https://github.com/koalaman/shellcheck#installing\n" >&2
-        printf "hint: you can disable shellcheck locally by setting \$MZDEV_NO_SHELLCHECK=1\n" >&2
+        echo -e "lint: $(red fatal:) unable to find \`shellcheck\` command on your system" >&2
+        echo -e "hint: https://github.com/koalaman/shellcheck#installing" >&2
+        echo -e "hint: you can disable shellcheck locally by setting \$MZDEV_NO_SHELLCHECK=1" >&2
         exit 1
     fi
     version=$(shellcheck --version | grep version: | grep -oE "[0-9]\.[0-9]\.[0-9]" || echo "0.0.0+unknown")
     if ! version_compat "0.7.0" "$version"; then
-        printf "lint: fatal: shellcheck v0.7.0+ is required\n" >&2
-        printf "hint: detected version %q\n" "$version" >&2
-        printf "hint: you can disable shellcheck locally by setting \$MZDEV_NO_SHELLCHECK=1\n" >&2
+        echo -e "lint: $(red fatal:) shellcheck v0.7.0+ is required" >&2
+        echo -e "hint: detected version %q" "$version" >&2
```

Comment:
echo doesn't do quote (`%q`) substitution, IMO this should still be `printf` in case version is empty.

Reply:
Oops, thanks, fixed.